packages/core/test/scp-cf/destination-accessor.spec.ts

```diff
@@ -79,11 +79,11 @@ describe('destination-accessor', () => {
       mockUserApprovedServiceToken();
 
       const httpMocks = [
-        mockInstanceDestinationsCall([], 200, subscriberServiceToken),
-        mockSubaccountDestinationsCall(oauthMultipleResponse, 200, subscriberServiceToken),
-        mockInstanceDestinationsCall([], 200, providerServiceToken),
-        mockSubaccountDestinationsCall([], 200, providerServiceToken),
-        mockSingleDestinationCall(oauthSingleResponse, 200, destinationName, userApprovedSubscriberServiceToken)
+        mockInstanceDestinationsCall(nock, [], 200, subscriberServiceToken),
```

Comment:
Why was this change necessary. It seems to me like a sensible default...

Reply:
Huh. Can you share more details here? Is that related to a version upgrade? I never noticed this problem before.

Reply:
No it is not related to an update. It is related to the previous hoisting. Situation before:
root
|---node_modules
|      |---nock
|---package_1
      |---node_modules
            |---ref_to_nock_root
|---package_2
      |---node_modules
            |---ref_to_nock_root

So when you did a `require(nock)` in package_1 or package_2 it was taking the same sources. Withtout the hoisting this has changed to:
root
|---package_1
      |---node_modules
            |---nock
|---package_2
      |---node_modules
            |---nock
So when you do a `require(nock)` in package_1 for example in the `destination-service-mocks.ts` of core/test-util and then in the `request-builder.spec.ts ` of the test-packages/integration-tests you load different source files. In principle this should not make a difference, but I found that there must be some encapsulation involved which breaks the matching on the first loaded nock. In other words the method `nock.activeMocks()` in package_1 do not show active mocks created by the second required nock in the  package_2 and when calls are executed the mocks are not triggered. 

Thats is why I pass

Reply:
Wow, that's evil... With that context this change makes a lot more sense.

Reply:
But couldn't that be solved by only having nock on the root level? According to (my understanding of) the node resolution algorithm, the root level node_module would be used and it would always be the same.

Reply:
technically yes, but I'm not a fan. You're essentially coupling the package to the project root. I am afraid that this a decision that has consequences that are not immediately obvious. Also it's something that you need to know when working with the project. If someone who's new to the project tries to run tests, then gets an error at runtime because nock isn't found, then looks at the package json and sees that nock isn't even listed as dependency and then adds it, I wouldn't blame them.

Reply:
I already discussed with Marika on this topic and I have no strong opinions. In principle it would be strict to have only linting etc. things on root and all package related stuff on the sub-packages. However this is also kind of inconvinient because something is needed everywhere so why putting it N times? 

My real Problem overall is: It should be consistent. Either put everything on root, or only the minimal stuff. Which one we chose I do not really care.

Reply:
I am in favor of putting it in the root except for dependencies that are run from the command line (there it has to be in the same context). I think dragging the dependency through the tests is worse than someone running `npm install` in a subpackage only. That can be covered by documentation and reviews.
I am also ok with doing this one we move to yarn.